yami_dec: remove --enable-capi from Makefile, to fix issue #394

wudping opened the pull request:
1, remove --enable-capi option from configure.ac, always compile libyami_capi.la.
2, remove all micro "ENABLE-CAPI" from code.

Signed-off-by: wudping dongpingx.wu@intel.com

dongpingx-wu committed:
yami_dec: remove --enable-capi from Makefile

1, remove --enable-capi option from configure.ac, always compile libyami_capi.la.
2, remove all micro "ENABLE-CAPI" from code.

Signed-off-by: wudping <dongpingx.wu@intel.com>

xuguangxin commented on encoder/vaapiencoder_h264.cpp:
```diff
@@ -721,6 +721,8 @@ VaapiEncoderH264::VaapiEncoderH264()
     m_videoParamAVC.enableDeblockFilter = true;
     m_videoParamAVC.deblockAlphaOffsetDiv2 = 2;
     m_videoParamAVC.deblockBetaOffsetDiv2 = 2;
+    m_videoParamAVC.sliceNum.iSliceNum = 1;
```
is any place use this? if not maybe we can remove this

wudping replied:
okay,  thanks @xuguangxin

xuguangxin commented:
+1

dongpingx-wu committed:
include: remove direciton interface

remove interface from include path in all the sorcecode.
header files in interface use include <> to include each other.

Signed-off-by: Jay Sorg <jay.sorg@intel.com>
Signed-off-by: wudping <dongpingx.wu@intel.com>

dongpingx-wu committed:
yamiversion: force version interface supporting C

add extern "C" into YamiVersion.h.

Signed-off-by: Jay Sorg <jay.sorg@intel.com>
Signed-off-by: wudping <dongpingx.wu@intel.com>

dongpingx-wu committed:
Yami.h: add two header files

add two header files Yami.h and YamiC.h for users to avoid
including too many headers when developing a app based on libyami;

Yami.h for CPP application;
YamiC.h for C application;

Signed-off-by: wudping <dongpingx.wu@intel.com>

wudping commented:
have integrated @jsorg71's patch #555

xuguangxin commented on Makefile.am:
```diff
@@ -17,8 +17,14 @@ if ENABLE_V4L2
 endif
 
+libyami_source_h = \
+	interface/libyami.h \
```
Please name it as Libyami.h just like other interface file

wudping replied:
okay, thanks @xuguangxin .

xuguangxin commented on interface/libyami.h:
```diff
@@ -0,0 +1,40 @@
+#include <VideoPostProcessInterface.h>
+
+
+#include <VideoDecoderCapi.h>
```
This should include in LibyamiC.h

wudping replied:
okay

xuguangxin commented on interface/libyami.h:
```diff
@@ -0,0 +1,40 @@
+
+#include <VideoCommonDefs.h>
+
+#include <VideoDecoderDefs.h>
```
Do we really ned this, Is it include in  `VideoDecoderHost.h`
I mean is it ok to just include VideoDecoderHost.h, VideoEncoderHost.h VideoPostProcessHost.h

wudping replied:
okay,

xuguangxin commented on interface/libyami.h:
```diff
@@ -0,0 +1,40 @@
+#include <VideoCommonDefs.h>
+
+#include <VideoDecoderDefs.h>
+#include <VideoDecoderHost.h>
```
Please change VideoCommonDefs.h and all other header file's include to <>

wudping replied:
okay

wudping replied:
but the libyami local file also include those header files, so it will have to scan the installed path when compiling libyami if their include are all <>.

xuguangxin commented on interface/LibyamiC.h:
```diff
@@ -0,0 +1,23 @@
+
+#ifndef LIBYAMI_C_H_
+#define LIBYAMI_C_H_
+
```
should have YamiVersion.h too

wudping replied:
thanks.

xuguangxin commented:
great! +1 for me

wudping commented:
modify Makefile.unittest for --enable-gtest.

xuguangxin commented on interface/YamiC.h:
```diff
@@ -0,0 +1,25 @@
+ * limitations under the License.
+ */
+
+#ifndef LIBYAMI_C_H_
```
changei to YAMI_C_H_

wudping replied:
okay, thanks @xuguangxin

chivakker commented:
The pull request lgtm.  It would be good to run an independent verification before merging.

wudping commented:
okay, thanks @chivakker .

jsorg71 commented:
+1

wudping commented:
covered by regression test

wudping commented:
thanks @jsorg71 .

xuguangxin commented:
merged as 26096137d99dc6af5a11e5484752d30f6f6552bd
